src/ledger/database/lmdb.zig

```diff
@@ -0,0 +1,716 @@
+const std = @import("std");
+const c = @import("lmdb");
+const sig = @import("../../sig.zig");
+const database = @import("lib.zig");
+
+const Allocator = std.mem.Allocator;
+
+const BytesRef = database.interface.BytesRef;
+const ColumnFamily = database.interface.ColumnFamily;
+const IteratorDirection = database.interface.IteratorDirection;
+const Logger = sig.trace.Logger;
+
+const key_serializer = database.interface.key_serializer;
+const value_serializer = database.interface.value_serializer;
+
+pub fn LMDB(comptime column_families: []const ColumnFamily) type {
+    return struct {
+        allocator: Allocator,
+        env: *c.MDB_env,
+        dbis: []const c.MDB_dbi,
+        path: []const u8,
+
+        const Self = @This();
+
+        pub fn open(allocator: Allocator, logger: Logger, path: []const u8) anyerror!Self {
```

Comment:
What's the reasoning for anyerror everywhere?

Reply:
I wrote a comptime function to enforce interface implementation by these databases. That code does not work if a functions has an inferred error return type that may differ depending on comptime parameters, because the return type is null when you do `@typeInfo` on the function.

I can improve some of these (like open) that do not use bincode. But for functions that call bincode, we'll need anyerror, unless I loosen up the interface enforcement a bit more.